core/dbt/compilation.py

```diff
@@ -82,16 +84,17 @@ def _generate_stats(manifest: Manifest):
         if _node_enabled(node):
             stats[node.resource_type] += 1
 
-    for source in manifest.sources.values():
-        stats[source.resource_type] += 1
-    for exposure in manifest.exposures.values():
-        stats[exposure.resource_type] += 1
-    for metric in manifest.metrics.values():
-        stats[metric.resource_type] += 1
-    for macro in manifest.macros.values():
-        stats[macro.resource_type] += 1
-    for group in manifest.groups.values():
-        stats[group.resource_type] += 1
+    # REVIEW: Why are these counted if disabled, when nodes in the general node
+    # collection are not?
```

Comment:
Poked into this -- I believe it's because the disabled mapping on the manifest stores a broader set of node types ([GraphMemberNode](https://github.com/dbt-labs/dbt-core/blob/6604b9ca31726098e3a8455e008ea12ef46bac7d/core/dbt/contracts/graph/nodes.py#L1611-L1616)) than the general node collection. For example, disabled exposures would be omitted from `manifest.exposures` and instead find themselves in `manifest.disabled` and so would not get included in the count. It looks like the `_node_enabled` is really just handling `NodeType.Test` resources which presumably find themselves in `manifest.nodes` even if they are disabled.

Reply:
That does make sense. I'll remove the comment.